mainapp/src/Component/UploadGraphModal.tsx

```diff
@@ -77,6 +82,46 @@ export function UploadGraphModal({ onClose, open, sourceName, indexAfterSuccess
 
         await writeLog(currentUser.login, "GraphManagement", "upload", sourceName);
 
+        if (graphUrl !== "") {
+            await uploadGraphByUrl(currentUser.token);
+        } else {
+            await uploadGraphByFile(currentUser.token);
+        }
+
+        if (indexAfterSuccess) {
+            try {
+                window.SearchUtil.generateElasticIndex(sourceName, { indexProperties: 1, indexNamedIndividuals: 1 }, () => {
+                    fetch(`/api/v1/ontologyModels?source=${sourceName}`, { method: "DELETE" })
+                        .then((_success) => {
+                            delete window.Config.ontologiesVocabularyModels[sourceName];
+                            window.UI.message(`${sourceName} was updated successfully`, true);
+                        })
+                        .catch((error) => {
+                            alert(error);
+                        });
+                });
+            } catch (error) {
+                console.error(error);
+                window.UI.message(`An error occurs during ${sourceName} update`, true);
+            }
+        }
+    };
+
+    const uploadGraphByUrl = async (userToken: string) => {
+        const formData = new FormData();
+        formData.append("url", graphUrl);
+        formData.append("source", sourceName);
+        // POST data
+        const res = await fetch(`${slsPyApiBaseUrl}api/v1/rdf/graphUrl`, { method: "post", headers: { Authorization: `Bearer ${userToken}` }, body: formData });
```

Comment:
We don't need the slsPyApi here

```suggestion
        const res = await fetch("/api/v1/rdf/graphUrl", { method: "post", headers: { Authorization: `Bearer ${userToken}` }, body: formData });
```